internal/pkg/propagation/stdlib.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package propagation
+
+import (
+	"go/types"
+	"strings"
+
+	"github.com/google/go-flow-levee/internal/pkg/utils"
+	"golang.org/x/tools/go/ssa"
+)
+
+// taintStdlibCall propagates taint through a static call to a standard
+// library function, provided that the function's taint propagation behavior
+// is known (i.e. the function has a summary).
+func (prop *Propagation) taintStdlibCall(call *ssa.Call, maxInstrReached map[*ssa.BasicBlock]int, lastBlockVisited *ssa.BasicBlock) {
+	summ, ok := funcSummaries[funcName(call)]
+	if !ok {
+		return
+	}
```

Comment:
How exhaustive is the current summary set?  Would it be better to explicitly enumerate the "no taint propagated" methods and indicate error here if an unemumerate function is found?

Reply:
For the set of packages I covered, I have good confidence that I have covered the relevant functions. For the packages I chose not to cover, I don't know if they may contain relevant functions. I could cover the remaining packages and add any new summaries in a future PR?

Reply:
I was more asking if, for the packages covered, if the functions included in the summaries constituted more than, say, 90% of the functions that were present in those packages.  I don't expect there to be much churn in the stdlib without us noticing, but it could be a good defensive measure to make sure we're not skipping over some functions without deliberately having excluded them and/or end up with a function signature that should be matching a summary but isn't due to a bug.

That investigation could also be opened as an Issue and rolled out as another PR, though.

Reply:
Hmm. From memory, I think for a good number of packages it was a significant fraction of the exported functions, but I would need to double check. Also we may want to monitor Go releases for new functions. Opened #293.